README.md

```diff
@@ -340,6 +327,61 @@ Note how this example responds to various types of navigations:
 
 Notice also how by passing through the `AbortSignal` found in `e.signal`, we ensure that any aborted navigations abort the associated fetch as well.
 
+#### Example: async transitions with special back/forward handling
+
+Sometimes it's desirable to handle back/forward navigations specially, e.g. reusing cached views by transitioning them onto the screen. This can be done by branching as follows:
+
+```js
+appHistory.addEventListener("navigate", e => {
+  // As before.
+  if (!e.canRespond || e.hashChange) {
+    return;
+  }
+
+  if (myFramework.currentPage) {
+    await myFramework.currentPage.transitionOut();
+  }
+
+  const isBackForward = appHistory.entries.includes(event.destination);
+  let { key } = event.destination;
+
+  if (isBackForward && myFramework.previousPages.has(key)) {
+    await myFramework.previousPages.get(key).transitionIn();
+  } else {
+    // This will probably result in myFramework storing the rendered page in myFramework.previousPages.
+    await myFramework.renderPage(event.destination);
+  }
+});
+```
+
+#### Restrictions on firing, canceling, and responding
+
+There are many types of navigations a given page can experience; see [this appendix](#appendix-types-of-navigations) for a full breakdown. Some of these need to be treated specially for the purposes of the navigate event.
+
+First, the following navigations **will not fire `navigate`** at all:
+
+- User-initiated [cross-document](#appendix-types-of-navigations) navigations via browser UI, such as the URL bar, back/forward button, or bookmarks.
+- [`document.open()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/open), which can strip off the fragment from the current document's URL.
+
+User-initiated navigations of this sort are outside the scope of the webpage, and can never be intercepted or prevented, even if they are to same-origin documents. On the other hand, we do allow the page to intercept user-initiated _same_-document navigations via browser UI, e.g. if the user changes the fragment component in the URL bar.
```

Comment:
This "even if they are to same-origin documents" only refers to `document.open()` right? If so, move it into that bullet, otherwise it's confusing because it looks like it's also possible in bullet 1

Reply:
No, this is meant to cover cases like:

1. You are on `https://example.com/foo`
2. The user types `https://example.com/bar` in the URL bar

Reply:
ah okay I got confused by same-origin same-document again :)